From 706a6abcc935750cc1a4eb3d41f3f3e95c813a4a Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Mon, 13 Nov 2017 15:49:38 +0800
Subject: [PATCH 2816/5242] MLK-16765-8 ARM64: dts: freescale: fsl-imx8qxp:
 enable PWRON key in dts

commit  160176b58a8ff388b5a9a9f516d87fcbf66a08c6 from
https://source.codeaurora.org/external/imx/linux-imx.git

Enable PWRON key in dts.

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
Reviewed-by: Peng Fan <peng.fan@nxp.com>
Reviewed-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
index 297c9d2..c2275a8 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
@@ -19,6 +19,7 @@
 #include <dt-bindings/soc/imx8_hsio.h>
 #include <dt-bindings/soc/imx8_pd.h>
 #include <dt-bindings/clock/imx8qxp-clock.h>
+#include <dt-bindings/input/input.h>
 #include <dt-bindings/pinctrl/pads-imx8qxp.h>
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/thermal/thermal.h>
@@ -2318,6 +2319,12 @@
 		};
 	};
 
+	sc_pwrkey: sc-powerkey {
+		compatible = "fsl,imx8-pwrkey";
+		linux,keycode = <KEY_POWER>;
+		wakeup-source;
+	};
+
 	wdog: wdog {
 		compatible = "fsl,imx8-wdt";
 	};
-- 
1.7.9.5

